addons/pos_sale/models/pos_order.py

```diff
@@ -50,57 +50,39 @@ def sync_from_ui(self, orders):
-            used_pos_lines = order.lines.sale_order_origin_id.order_line.pos_order_line_ids.ids
-            lines = order.lines.filtered(
-                lambda l: (
-                    l.id not in used_pos_lines
-                    and l.product_id == order.config_id.down_payment_product_id
-                    and l.qty != 0
-                    and (
-                        l.sale_order_origin_id
-                        or l.refunded_orderline_id.sale_order_origin_id
-                    )
-                )
-            )
-            for line in lines:
-                sale_lines = line.sale_order_origin_id.order_line or line.refunded_orderline_id.sale_order_origin_id.order_line
-                sale_order_origin = line.sale_order_origin_id or line.refunded_orderline_id.sale_order_origin_id
-                if not any(line.display_type and line.is_downpayment for line in sale_lines):
-                    self.env['sale.order.line'].create(
-                        self.env['sale.advance.payment.inv']._prepare_down_payment_section_values(sale_order_origin)
-                    )
-                order_reference = line.name
-
-                if order.partner_id.lang and order.partner_id.lang != line.env.lang:
-                    line = line.with_context(lang=order.partner_id.lang)
-
-                sale_order_line_description = _("Down payment (ref: %(order_reference)s on \n %(date)s)", order_reference=order_reference, date=format_date(line.env, line.order_id.date_order))
-                sale_line = self.env['sale.order.line'].create({
-                    'order_id': sale_order_origin.id,
-                    'product_id': line.product_id.id,
-                    'price_unit': line.price_unit,
-                    'product_uom_qty': 0,
-                    'tax_ids': [(6, 0, line.tax_ids.ids)],
-                    'is_downpayment': True,
-                    'discount': line.discount,
-                    'sequence': sale_lines and sale_lines[-1].sequence + 2 or 10,
-                    'name': sale_order_line_description
-                })
-                line.sale_order_line_id = sale_line
-
-            so_lines = order.lines.mapped('sale_order_line_id')
-
-            if order.state != 'draft':
-                # confirm the unconfirmed sale orders that are linked to the sale order lines
-                sale_orders = so_lines.mapped('order_id')
+        AccountTax = self.env['account.tax']
+        pos_orders = self.browse([o['id'] for o in data["pos.order"]])
+        for pos_order in pos_orders:
+            # TODO: the way to retrieve the sale order in not consistent... is it a bad code or intended?
+            used_pos_lines = pos_order.lines.sale_order_origin_id.order_line.pos_order_line_ids
+            downpayment_pos_order_lines = pos_order.lines.filtered(lambda line: (
+                line not in used_pos_lines
+                and line.product_id == pos_order.config_id.down_payment_product_id
+            ))
```

Comment:
```suggestion
            downpayment_pos_order_lines = pos_order.lines.filtered(lambda line: line.product_id == pos_order.config_id.down_payment_product_id)
```

It looks unnecessary. Can you try removing it? Let see if the tests are okay with it.

Reply:
@caburj I prefer to avoid to change that part. My branch is huge enough and even it doesn't break any test, that looks like an unecessary risk that is not related to this task. Let's change this after the merge ;)